### ISSUE TITLE
Logout says delete identity

### ISSUE THREAD

**Issue by weex** [innovatario/yondar-mono#5]
Really cool idea! Location plus nostr I think has a lot of potential. As a first time user, I clicked on a button to login. I can't get back to that page so I'm not sure what it was called but it was a few messages into the chat flow.

Once it became clear that I was creating a new Nostr identity, I wanted to logout and do NIP-07 login. Is there a reason an identity has to be deleted on logout? It's still not clear if I'm logged in or not.

Sorry for this issue being all over the place but I didn't want to spam the tracker with a lot of little ones. Looking forward to see how this develops!

**Comment by arkin0x** [comment]
Hello @weex, 

The application acts differently depending on if there is an identity stored in localStorage. The identity consists of a public key and profile metadata loaded from nostr.

If the identity was created by generating a new private key, then that key will also be encrypted in localStorage.

In order to log out and make the application treat you as if you are logged out, you must clear these things from localStorage. Once cleared, they are effectively deleted.

You can export your private key if you generated it, which allows you to save that identity before you delete it from Yondar.

We could make it so the identity is not deleted when you log out, but I don't really see a good reason to do that. Logging out means you want the application to not recognize you. What would be the benefit of logging out and not deleting the identity?

Thanks!

**Comment by arkin0x** [comment]
To clarify, the identity may be deleted from Yondar's localStorage but it can't be deleted from nostr. If you backed up your generated private key, you won't lose anything. I'm amenable to changing this wording to avoid confusion.

**Comment by weex** [comment]
> What would be the benefit of logging out and not deleting the identity?

When I clicked whatever button it was I didn't mean to generate a new identity but once it's been made I don't mind having it to play around with. My reason to logout was partly as a tester wanting to see that first screen again, and then as a user wanting to login with my normal nostr identity. When it said "delete identity" I as a user started thinking about tradeoffs and didn't have enough info to evaluate.

I suppose a better way would have been logout then give me the choice next time to use the generated identity (I did save the password) or nip-07 or whatever.
